### PR TITLE
Tools  - Part 2

### PR DESCRIPTION
**Pull request by sri2k1us**
- on Top #247 
- Tool Details
image

image

-iPhone 5
image

### CODE REVIEW

File: src/components/tools/details/Drawer.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    const drawerId = build(baseId, ids.DETAILS_DRAWER);
+    const infoTabId = build(drawerId, ids.INFO_TAB);
+    const paramsTabId = build(drawerId, ids.APPS_USING_TOOL);
+
+    const { isFetching: isInfoFetching, error: infoFetchError } = useQuery({
+        queryKey: [TOOL_DETAILS_QUERY_KEY, { id: selectedTool?.id }],
+        queryFn: getToolDetails,
+        config: {
+            enabled: selectedTool !== null && selectedTool !== undefined,
```
Comment (psarando) [inline src/components/tools/details/Drawer.js:91]:
```suggestion
            enabled: !!selectedTool,
```

File: src/components/tools/details/Drawer.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            onSuccess: setDetails,
+        },
+    });
+
+    const { isFetching: isAppsFetching, error: appsFetchError } = useQuery({
+        queryKey: [APPS_USING_QUERY_KEY, { id: selectedTool?.id }],
+        queryFn: getAppsUsed,
+        config: {
+            enabled: selectedTool !== null && selectedTool !== undefined,
```
Comment (psarando) [inline src/components/tools/details/Drawer.js:100]:
```suggestion
            enabled: !!selectedTool,
```

File: src/components/tools/details/Drawer.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                />
+            </Tabs>
+            <DETabPanel
+                tabId={infoTabId}
+                value={TABS.toolDetails}
+                selectedTab={selectedTab}
+            >
+                <ToolDetails
+                    tool={details}
+                    baseDebugId="toolDetails"
```
Comment (psarando) [inline src/components/tools/details/Drawer.js:156]: This ID should probably come from the `ids` file.

File: src/components/tools/details/Drawer.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    const classes = useStyles();
+    const { t } = useTranslation("tools");
+    const { selectedTool, open, onClose, baseId } = props;
+    const [selectedTab, setSelectedTab] = useState(TABS.toolDetails);
+    const [details, setDetails] = useState();
+    const [apps, setApps] = useState();
+
+    const drawerId = build(baseId, ids.DETAILS_DRAWER);
+    const infoTabId = build(drawerId, ids.INFO_TAB);
+    const paramsTabId = build(drawerId, ids.APPS_USING_TOOL);
```
Comment (psarando) [inline src/components/tools/details/Drawer.js:85]: It might make more sense to name this `appsTabId`, or something like that.

**Review by psarando — APPROVED** [review]
LGTM 👍 
I just had a few minor suggestions ✨

File: src/components/tools/details/Drawer.js (hunk 4 above)
Comment (sri2k1us) [reply src/components/tools/details/Drawer.js:85]: Oops..Copy Pasta issue.

File: src/pages/tools.js [hunk 5]
```diff
@@ -1,8 +1,51 @@
...
+import { getLocalStorage } from "components/utils/localStorage";
+import Listing from "components/tools/listing/Listing";
+
+import NavigationConstants from "common/NavigationConstants";
 
 export default function Tools() {
-    return <Listing baseId="tools" />;
+    const router = useRouter();
+    const query = router.query;
+    const { t } = useTranslation("tools");
```
Comment (ash3rz) [inline src/pages/tools.js:22]: Github actions warning that this isn't used

File: src/components/tools/details/ToolDetails.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            />
+        );
+    }
+
+    if (!tool && !isInfoFetching && !infoFetchError) {
+        return null;
+    }
+
+    return (
+        <Paper id={baseDebugId} style={{ padding: 5 }}>
```
Comment (ash3rz) [inline src/components/tools/details/ToolDetails.js:33]: Looking at the screenshots, how does it look without the wrapping `Paper`?  It looks a little cramped to me with the paper added, not sure what others think.

**Review by ash3rz — APPROVED** [review]
LGTM, just had some minor comments :shipit:

File: src/components/tools/details/ToolDetails.js (hunk 6 above)
Comment (sri2k1us) [reply src/components/tools/details/ToolDetails.js:33]: Removing `Paper` makes it better. So I will go ahead and do that.
image

**Comment by sri2k1us** [comment]
Thanks for the reviews.

**Pull request merged**